debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9IndexableObjectHelper.java

```diff
@@ -143,39 +218,71 @@ public static long discontiguousHeaderSize()
 	{
 		if (mixedReferenceMode) {
 			if (compressObjectReferences) {
-				return VoidPointer.cast((J9IndexableObjectContiguousCompressedPointer.cast(objPointer)).dataAddr());
+				return VoidPointer.cast(J9IndexableObjectWithDataAddressContiguousCompressedPointer.cast(objPointer).dataAddr());
 			}
-			return VoidPointer.cast((J9IndexableObjectContiguousFullPointer.cast(objPointer)).dataAddr());
+			return VoidPointer.cast(J9IndexableObjectWithDataAddressContiguousFullPointer.cast(objPointer).dataAddr());
 		}
-		return VoidPointer.cast((J9IndexableObjectContiguousPointer.cast(objPointer)).dataAddr());
+		return VoidPointer.cast(J9IndexableObjectWithDataAddressContiguousPointer.cast(objPointer).dataAddr());
 	}
 
 	/**
-	 * @param objPointer the discontiguous array object who's dataAddr field we are accessing
-	 * @throws CorruptDataException If there's a problem accessing the indexable object dataAddr field
-	 * @throws NoSuchFieldException If the indexable object dataAddr field does not exist on the build that generated the core file
+	 * Get the dataAddr field for the discontiguous array object.
+	 *
+	 * @param objPointer the discontiguous array object whose dataAddr field we are accessing
+	 * @throws CorruptDataException if there's a problem accessing the indexable object dataAddr field
+	 * @throws NoSuchFieldException if the indexable object dataAddr field does not exist on the build that generated the core file
 	 */
 	public static VoidPointer getDataAddrForDiscontiguous(J9IndexableObjectPointer objPointer) throws CorruptDataException, NoSuchFieldException
 	{
 		if (mixedReferenceMode) {
 			if (compressObjectReferences) {
-				return VoidPointer.cast((J9IndexableObjectDiscontiguousCompressedPointer.cast(objPointer)).dataAddr());
+				return VoidPointer.cast(J9IndexableObjectWithDataAddressDiscontiguousCompressedPointer.cast(objPointer).dataAddr());
 			}
-			return VoidPointer.cast((J9IndexableObjectDiscontiguousFullPointer.cast(objPointer)).dataAddr());
+			return VoidPointer.cast(J9IndexableObjectWithDataAddressDiscontiguousFullPointer.cast(objPointer).dataAddr());
 		}
-		return VoidPointer.cast((J9IndexableObjectDiscontiguousPointer.cast(objPointer)).dataAddr());
+		return VoidPointer.cast(J9IndexableObjectWithDataAddressDiscontiguousPointer.cast(objPointer).dataAddr());
+	}
+
+	/**
+	 * Get the dataAddr field for the indexable object.
+	 *
+	 * @param objPointer the array object whose dataAddr field we are accessing
+	 * @throws CorruptDataException if there's a problem accessing the indexable object dataAddr field
+	 * @throws NoSuchFieldException if the indexable object dataAddr field does not exist on the build that generated the core file
+	 */
+	public static VoidPointer getDataAddrForIndexable(J9IndexableObjectPointer objPointer) throws CorruptDataException, NoSuchFieldException
+	{
+		return isInlineContiguousArraylet(objPointer)
```

Comment:
This could use `ObjectModel.isInlineContiguousArraylet()` instead of the simple forwarding function in this class, which can be deleted.